rclcpp/src/rclcpp/graph_listener.cpp

```diff
@@ -142,6 +145,11 @@ GraphListener::run_loop()
     std::lock_guard<std::mutex> nodes_lock(node_graph_interfaces_mutex_, std::adopt_lock);
     // Ensure that the context doesn't go out of scope.
     auto parent_context = parent_context_.lock();
+    if (!parent_context) {
+      // the parent context may be destroyed before this loop is stopped.
+      // in that case, just return silently.
```

Comment:
nitpick: if a comment has punctuation then it should be a sentence and should have capitalization

```c++
// this is ok
// This is also ok.
// some_variable_name at the beginning makes a lack of capitalization ok too.

// this is weird.
// Weird because not a sentence, actually sentence fragment.
// not weird because not a sentence, so sentence fragments are ok
```